drivers/ng_at86rf2xx/ng_at86rf2xx_getset.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ng_at86rf2xx
+ * @{
+ *
+ * @file
+ * @brief       Getter and setter functions for the AT86RF2xx drivers
+ *
+ * @author      Thomas Eichinger <thomas.eichinger@fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "ng_at86rf2xx.h"
+#include "ng_at86rf2xx_internal.h"
+#include "ng_at86rf2xx_registers.h"
+#include "periph/spi.h"
+
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+
+
+static const int16_t tx_pow_to_dbm[] = {3, 3, 2, 2, 1, 1, 0,
+                                        -1, -2, -3, -4, -5, -7, -9, -12, -17};
+
+static const uint8_t dbm_to_tx_pow[] = {0x0f, 0x0f, 0x0f, 0x0e, 0x0e, 0x0e,
+                                        0x0e, 0x0d, 0x0d, 0x0c, 0x0c, 0x0b,
+                                        0x0b, 0x0a, 0x09, 0x08, 0x07, 0x06,
+                                        0x05, 0x03, 0x00};
+
+uint16_t ng_at86rf2xx_get_addr_short(ng_at86rf2xx_t *dev)
+{
+    return dev->addr_short;
+}
+
+void ng_at86rf2xx_set_addr_short(ng_at86rf2xx_t *dev, uint16_t addr)
+{
+    dev->addr_short = addr;
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__SHORT_ADDR_0, (addr >> 8));
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__SHORT_ADDR_1, (uint8_t)addr);
+}
+
+uint64_t ng_at86rf2xx_get_addr_long(ng_at86rf2xx_t *dev)
+{
+    return dev->addr_long;
+}
+
+void ng_at86rf2xx_set_addr_long(ng_at86rf2xx_t *dev, uint64_t addr)
+{
+    dev->addr_long = addr;
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__IEEE_ADDR_0, addr >> 56);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__IEEE_ADDR_1, addr >> 48);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__IEEE_ADDR_2, addr >> 40);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__IEEE_ADDR_3, addr >> 32);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__IEEE_ADDR_4, addr >> 24);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__IEEE_ADDR_5, addr >> 16);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__IEEE_ADDR_6, addr >> 8);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__IEEE_ADDR_7, addr);
+}
+
+uint8_t ng_at86rf2xx_get_chan(ng_at86rf2xx_t *dev)
+{
+    uint8_t res = ng_at86rf2xx_reg_read(dev, NG_AT86RF2XX_REG__PHY_CC_CCA);
+    return (res & NG_AT86RF2XX_PHY_CC_CCA_MASK__CHANNEL);
+}
+
+void ng_at86rf2xx_set_chan(ng_at86rf2xx_t *dev, uint8_t channel)
+{
+    uint8_t tmp;
+
+    if (channel < NG_AT86RF2XX_MIN_CHANNEL
+        || channel > NG_AT86RF2XX_MAX_CHANNEL) {
+        return;
+    }
+    tmp = ng_at86rf2xx_reg_read(dev, NG_AT86RF2XX_REG__PHY_CC_CCA);
+    tmp &= ~(NG_AT86RF2XX_PHY_CC_CCA_MASK__CHANNEL);
+    tmp = (channel & NG_AT86RF2XX_PHY_CC_CCA_MASK__CHANNEL);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__PHY_CC_CCA, tmp);
+}
+
```

Comment:
@kaspar030, should we add the option for select the channel here as a get/set?

    #ifndef MODULE_NG_AT86RF212B
    void ng_at86rf2xx_set_freq(ng_at86rf2xx_t *dev, ng_at86rf2xx_freq_t freq)
    {
        uint8_t tmp;
        if (freq != FREQ_915MHZ && freq != FREQ_868MHZ) {
            return;
        }
        tmp = ng_at86rf2xx_reg_read(dev, NG_AT86RF2XX_REG__TRX_CTRL_2);
        tmp &= ~(NG_AT86RF2XX_TRX_CTRL_2_MASK__SUB_MODE);
        tmp |= freq;
        ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__TRX_CTRL_2, tmp);
    }
    ng_at86rf2xx_freq_t ng_at86rf2xx_get_freq(ng_at86rf2xx_t *dev)
    {
        uint8_t tmp;
        tmp = ng_at86rf2xx_reg_read(dev, NG_AT86RF2XX_REG__TRX_CTRL_2);
        return (ng_at86rf2xx_freq_t)(tmp & NG_AT86RF2XX_TRX_CTRL_2_MASK__SUB_MODE);
    }
    #endif 

And an enum in drivers/ng_at86rf2xx/include/ng_at86rf2xx_registers.h

    #ifndef MODULE_NG_AT86RF212B
    typedef enum {
        FREQ_915MHZ       =  NG_AT86RF2XX_TRX_CTRL_2_MASK__SUB_MODE,
        FREQ_868MHZ       = 0x00 ,
    } ng_at86rf2xx_freq_t;
    #endif

What do you think? (i haven't tried it)